x/proof/keeper/msg_server_create_claim.go

```diff
@@ -73,13 +80,58 @@ func (k msgServer) CreateClaim(
 		RootHash:        msg.GetRootHash(),
 	}
 
-	// TODO_BLOCKER(@Olshansk): check if this claim already exists and return an
-	// appropriate error in any case where the supplier should no longer be able
-	// to update the given proof.
+	_, isExistingClaim = k.Keeper.GetClaim(ctx, claim.GetSessionHeader().GetSessionId(), claim.GetSupplierAddress())
+
 	k.Keeper.UpsertClaim(ctx, claim)
 
 	logger.Info("created new claim")
 
+	// NB: Don't return these errors, it should not prevent the MsgCreateProofResopnse
```

Comment:
Same as proof, I believe we should fail if `claim.GetNumRelays` or `claim.GetNumComputeUnits` fail.